File: SpringBootDev/src/main/java/org/springbootdev/modules/develop/entity/TableInfo.java

package org.springbootdev.modules.develop.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;
import org.springbootdev.core.mp.base.BaseEntity;

/**
 * 实体类
 *
 * @author zhaobohao
 * @since 2019-12-17
 */
@Data
@TableName("db_table_info")
@EqualsAndHashCode(callSuper = true)
@ApiModel(value = "TableInfo对象", description = "TableInfo对象")
@Accessors(chain = true)
public class TableInfo extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /**
     * 主键
     */
    @ApiModelProperty(value = "主键")
    @TableId(value = "id", type = IdType.AUTO)
  private Long id;
    /**
     * 名称
     */
    @ApiModelProperty(value = "名称")
    private String name;
    /**
     * 字段类型
     */
    @ApiModelProperty(value = "字段类型")
    private String typeKey;
    /**
     * 字段类型值
     */
    @ApiModelProperty(value = "字段类型值")
    private String typeValue;
    /**
     * 是否为空 1为空2不为空
     */
    @ApiModelProperty(value = "是否为空 1为空2不为空")
    private Integer isEmpty;
    /**
     * 默认值 
     */
    @ApiModelProperty(value = "默认值 ")
    private String defaultValue;
    /**
     * 注释说明
     */
    @ApiModelProperty(value = "注释说明")
    private String comment;
    /**
     * 父级菜单
     */
    @ApiModelProperty(value = "父级菜单")
    private Long parentId;
    /**
     * 1表，2字段
     */
    @ApiModelProperty(value = "1表，2字段")
    private Integer category;
    /**
     * 数据库id
     */
    @ApiModelProperty(value = "数据库id")
    private Long dbInstanceId;
    /**
     * 是否是叶子节点，0是，1不是
     */
    @ApiModelProperty(value = "是否是叶子节点，0是，1不是")
    private Integer isLeaf;


}

File: SpringCloudDev/springcloud-vue-ui/src/entitys/tableinfo.js
export default function tableinfo() {
  return {
    current: 1,
    size: 20,
    isNewRecord: false,
    query: {
      id: undefined,
      // 名称
      name: undefined,
      // 字段类型
      typeKey: undefined,
      // 字段类型值
      typeValue: undefined,
      // 是否为空
      isEmpty: undefined,
      // 默认值
      defaultValue: undefined,
      // 注释说明
      comment: undefined,
      // 表ID
      parentId: undefined,
      // 数据类型
      category: undefined,
      // 数据库
      dbInstanceId: undefined,
      // 叶子节点
      isLeaf: undefined
    },
    rules: {
      name: [{
        required: true,
        message: '请输入名称',
        trigger: 'blur'
      }, {
        min: 5,
        max: 30,
        message: '长度在 5 到 30 个字符'
      }],
      typeKey: [{
        required: true,
        message: '请输入字段类型',
        trigger: 'blur'
      }],
      typeValue: [{
        required: true,
        message: '请输入字段类型值',
        trigger: 'blur'
      }],
      isEmpty: [{
        required: true,
        message: '请输入是否为空',
        trigger: 'blur'
      }],
      comment: [{
        required: true,
        message: '请输入注释说明',
        trigger: 'blur'
      }]

    }
  }
}
